### ISSUE TITLE
is it possible to modify dragonfly name? 

### ISSUE THREAD

**Issue by j4rv1s1** [DragonComputer/Dragonfire#64]
hi guys first of all congrats for this awesome project, finally something smarter. i'm here just to ask you if it's possible to modify dragonfly wake up words and name,if so in which file?  thank you!

**Comment by mertyildiran** [comment]
@j4rv1s1 thanks! :blush: No, it's not currently possible but it can be with a few modifications under [this block](https://github.com/DragonComputer/Dragonfire/blob/master/dragonfire/__init__.py#L212-L218) and by adding a new block to catch and handle renaming task.

I believe you can do this work by yourself and contribute to this project. Feel free to send pull requests. :trophy:

**Comment by Immortalin** [comment]
I think the linked block is out of date

**Comment by mertyildiran** [comment]
@Immortalin yeah, you're right. Actually almost every occurrence of `dragonfire` as a string in the code base must be parameterized to achieve this feature. We have a config file which we load with [this line](https://github.com/DragonComputer/Dragonfire/blob/81e4e2799d90dd3f6bf48c0ad263782237cb3391/dragonfire/__init__.py#L176) so we can store the custom name there. Also [we have a logic](https://github.com/DragonComputer/Dragonfire/blob/81e4e2799d90dd3f6bf48c0ad263782237cb3391/dragonfire/learn.py#L144-L145) to detect statements like *YOUR NAME IS SCARLET* (though I'm not happy with the current state of the code base) so with the combination of `h.check_deps_contains("your name")` and [subject extraction](https://github.com/DragonComputer/Dragonfire/blob/81e4e2799d90dd3f6bf48c0ad263782237cb3391/dragonfire/learn.py#L86-L105), it's achievable.

The code base's itself requires a lot of refactoring. @Immortalin if you are willing to help then feel free to refactor the code base. Because it has priority...

Also I'm planning to replace the spaCy calls in [__init__.py](https://github.com/DragonComputer/Dragonfire/blob/master/dragonfire/__init__.py) with the [Intent Classification of DeepPavlov.ai](https://demo.deeppavlov.ai/#/en/intent) so maybe you would like to help this project to get rid of gazillion `if`s in [__init__.py](https://github.com/DragonComputer/Dragonfire/blob/master/dragonfire/__init__.py)?

**Comment by EllaLoves** [comment]
You can rename a dragon by simply using a name tag!
Never noticed it was this simple until now
